src/effects/backends/builtin/pitchshifteffect.cpp

```diff
@@ -61,12 +69,34 @@ EffectManifestPointer PitchShiftEffect::getManifest() {
     pitch->setNeutralPointOnScale(0.0);
     pitch->setRange(-1.0, 0.0, 1.0);
 
+    EffectManifestParameterPointer semitonesMode = pManifest->addParameter();
+    semitonesMode->setId("semitonesMode");
+    semitonesMode->setName(QObject::tr("Semitones"));
+    semitonesMode->setShortName(QObject::tr("Semitones"));
+    semitonesMode->setDescription(QObject::tr(
+            "Change pitch in semitone-steps instead of smoothly."));
+    semitonesMode->setValueScaler(EffectManifestParameter::ValueScaler::Toggle);
+    semitonesMode->setUnitsHint(EffectManifestParameter::UnitsHint::Unknown);
+    semitonesMode->setRange(0, 1, 1);
+
+    EffectManifestParameterPointer formant = pManifest->addParameter();
+    formant->setId("formant");
+    formant->setName(QObject::tr("Formant"));
```

Comment:
It looks like RB "engine fine" actually supports to shift the formants. If we consider this, the name "Formant" is better used for the formant rotary knob. This one is "Formant Preserving" in case we use the LV2 variant. Is the length still OK?

Reply:
Yeah, in the R3 version the formant can be shifted using a knob. With introducing the R3 version in the Pitch shift effect, this option of course could be added and if it would be, the identical toggle name will be the problem too. On the other hand, the "Formant Preserving" name is too wide, see the print screen:

![image](https://user-images.githubusercontent.com/84876095/185974426-f9350bde-0a6c-4242-84ee-adc313cc24d3.png)

It looks like the maximum number of characters is about 10.

Reply:
Then just name it formant and put the detailed explanation in the tooltip imo.

Reply:
Ok, but at least the Id should not be "formant". Since the name can be changed at any time, but not the ID.

Reply:
Okay, that looks good to me.